### ISSUE TITLE
Dynamicform - New items are always created with the default content type if the list has multiple content types

### ISSUE THREAD

**Issue by mgitta** [pnp/sp-dev-fx-controls-react#1392]
#### Category
[ ] Enhancement

[X ] Bug

[ ] Question

#### Version

Please specify what version of the library you are using: [    3.12.0     ]

#### Expected / Desired Behavior / Question
The site has a list called List1. List1 has multiple content types, CT1, CT2 and CT3, the default content type is CT1. All content types have different fields added to them. When the user opens the new form, and creates a new item, the new item's CT matches the one the user chooses on the sharepoint interface.

#### Observed Behavior
The content type of the created item is always the default content type CT1, unless I set the listItem's content type in the onBeforeSubmit event.

#### Steps to Reproduce
I have a list called List1. List1 has multiple content types, CT1, CT2 and CT3, the default content type is CT1. I created a custom form for the list with DynamicForm. The listid and contenttypeid is passed to the Dynamicform. The new item will have the default content type.
```
 <DynamicForm
        context={this.props.context as any}
        listId={this.props.context.list.guid.toString()}
        contentTypeId={this.props.context.contentType.id}
        listItemId={this.props.context.itemId}
        onCancelled={this.props.onClose}
        onBeforeSubmit={this._onBeforeSubmit}
        onSubmitError={this._onSubmitError}
        onSubmitted={this._onSubmitted}
        returnListItemInstanceOnSubmit={true}
      />
  
 private _onBeforeSubmit = async (listItemData: any): Promise<boolean> => {
    //set the content type, because it has a bug - if the ct is not set manually, then for creation it will create the item with the default ct.
    listItemData.ContentTypeId = this.props.context.contentType.id;
....
}
```

**Comment by martinlingstuyl** [comment]
Thank you @mgitta for the `private _onBeforeSubmit = async (listItemData: any): Promise<boolean> => { }` workaround. It's a life saver.

@AJIXuMuK, when can we expect the next release?

**Comment by NishkalankBezawada** [comment]
Hello @SamueleFurnari, are you still intending to create a new pull request for this?

Thanks,
Nishkalank

**Comment by michaelmaillot** [comment]
Hi @mgitta,

Did you get a chance to retry with latest version (3.15.0)? It seems that a fix has been made.

**Comment by NishkalankBezawada** [comment]
Hi @michaelmaillot, I am trying to reproduce the issue with latest version, will keep you posted.

Thanks,
Nishkalank

**Comment by NishkalankBezawada** [comment]
Hello @michaelmaillot, 

I have reproduced the issue with the latest version, and the issue still persists. 

I will be creating a PR on this as soon as i could. 

Thanks,
Nishkalank Bezawada

**Comment by michaelmaillot** [comment]
Closing as fixed in 3.16.0
